astropy/cosmology/core.py

```diff
@@ -1752,6 +1753,64 @@ def __init__(self, H0, Om0, Tcmb0=0, Neff=3.04,
                                            self._nmasslessnu,
                                            self._nu_y_list)
 
+    def _elliptic_comoving_distance_z1z2(self, z1, z2):
+        """ Comoving line-of-sight distance in Mpc between objects at
+        redshifts z1 and z2.
+
+        The comoving distance along the line-of-sight between two
+        objects remains constant with time for objects in the Hubble
+        flow.
+
+        For Omega_radiation = 0 (T_CMB = 0; massless neutrinos)
+        the comoving distance can be directly calculated as an elliptic integral.
+        See
+            Feige, 1992, Astron. Nachr., 313, 139.
+            Eisenstein, 1997, arXiv:9709054v2
+            Mészáros & Řípai 2013, A&A, 556, A13.
+        and a useful summary in
+            Baes, Camps, Van De Putte, 2017, MNRAS, 468, 927.
+
+        Parameters
+        ----------
+        z1, z2 : array-like, shape (N,)
+          Input redshifts.  Must be 1D or scalar.
+
+        Returns
+        -------
+        d : `~astropy.units.Quantity`
+          Comoving distance in Mpc between each input redshift.
+        """
+        if isiterable(z1):
+            z1 = np.asarray(z1)
+            z2 = np.asarray(z2)
+            if z1.shape != z2.shape:
+                msg = "In _comoving_distance_z1z2: z1 and z2 have different lengths"
+                raise ValueError(msg)
+
+        s = ((1-self._Om0)/self._Om0) ** (1/3)
+        prefactor = self._hubble_distance / sqrt(s*self._Om0)
+        return prefactor * (self._T_legendre(s/(1+z1)) - self._T_legendre(s/(1+z2)))
+
+    def _T_legendre(self, x):
+        """ Compute T(x) using Legendre elliptical integrals of the first kind.
+
+        T(x) = 3^{-\\frac{1}{4}}
+               F\\left(arccos\\left(\\frac{1+(1-\\sqrt{3}x}{1+(1+\\sqrt{3})x}\\right),
+                      \\cos\\frac{\\pi}{12}\\right)
+        where
+        F(phi, m) = int_0^phi {1/sqrt(1 - m \\sin^2{t})} dt
+        """
+
+        from scipy.special import ellipkinc
+        F = ellipkinc
+        # math.sqrt is several times faster than np.sqrt for scalars
+        phi = np.arccos((1 + (1-sqrt(3))*x) /
+                        (1 + (1+sqrt(3))*x))
+        k = np.cos(np.pi/12)
+        m = k*k  # np.cos(np.pi/12)**2 == 1/2 + math.sqrt(3)/4
```

Comment:
Please just hardwire the value with the comment rather than recomputing each time.  And in that case you can omit the packages from the comment:

m = 0.9330127018922193  # cos(pi/12)**2